### PR TITLE
docs: Added missing explanation of implements OnInit

### LINKED ISSUE #51586: No mention or explanation of `implements OnInit`
### Describe the problem that you experienced

Under `step 2` of [View product details](https://angular.io/start/start-routing#view-product-details) section, the class `ProductDetailsComponent` implements `OnInit`, however, there is no mention of adding `implements OnInit` in the documentation.

```ts
export class ProductDetailsComponent implements OnInit {

  product: Product | undefined;
  /* ... */
}
```

### Enter the URL of the topic with the problem

https://angular.io/start/start-routing#view-product-details

### Describe what you were looking for in the documentation

There should be some mention and explanation of `implements OnInit` at this step.

### Describe the actions that led you to experience the problem

I'm following along with the official Angular documentation to learn Angular.

I have not skipped any steps, and have started from the very begining.

### Describe what you want to experience that would fix the problem

Please mention this change to the file `src/app/product-details/product-details.component.ts`.

### Add a screenshot if that helps illustrate the problem

image

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true
N/A.
```

### Provide any additional information here in as much as detail as you can

_No response_

### PR DESCRIPTION
**Pull request by srinathsree66**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

closes: #51586

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/start/start-routing.md [hunk 1]
```diff
@@ -60,11 +60,11 @@ The Angular Router displays components based on the browser's URL and [your defi
 
 In this section, you'll use the Angular Router to combine the `products` data and route information to display the specific details for each product.
 
-1.  In `product-details.component.ts`, import `ActivatedRoute` from `@angular/router`, and the `products` array from `../products`.
+1.  In `product-details.component.ts`, import `ActivatedRoute` from `@angular/router`, import `ngOnInit()` from `@angular/core`, and the `products` array from `../products`.
```
Comment (Yogu) [inline aio/content/start/start-routing.md:63]:
```suggestion
1.  In `product-details.component.ts`, import `ActivatedRoute` from `@angular/router`, import `OnInit` from `@angular/core`, and the `products` array from `../products`.
```

Comment (srinathsree66) [reply aio/content/start/start-routing.md:63]: Updated thank you.

Comment (JeanMeche) [reply aio/content/start/start-routing.md]: Can you please squash that additional commit ?

Comment (srinathsree66) [reply aio/content/start/start-routing.md]: Done

**Review by jessicajaniuk — APPROVED** [review]
LGTM

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit b6844a90564fc7095642ca3c8ee247c78677c4e5.

**Pull request closed without merging**